syntax = "proto3";

package pslx;

// the next will be 2
enum ModeType {
    TEST = 0;
    PROD = 1;
}

// the next will be 2
enum SortOrder {
    ORDER = 0;
    REVERSE = 1;
}

// the next will be 3
enum DataModelType {
    DEFAULT = 0;
    BATCH = 1;
    STREAMING = 2;
}

// the next will be 3
enum InstantMessagingType {
    SLACK = 0;
    ROCKETCHAT = 1;
    TEAMS = 2;
}

// the next will be 5
enum DiskLoggerLevel {
    INFO = 0;
    DEBUG = 1;
    WARNING = 2;
    NOTSET = 3;
    ERROR = 4;
}

// the next will be 5
enum StorageType {
    DEFAULT_STORAGE = 0;
    PARTITIONER_STORAGE = 1;
    FIXED_SIZE_STORAGE = 2;
    PROTO_TABLE_STORAGE = 3;
    SHARDED_PROTO_TABLE_STORAGE = 4;
}

// the next will be 2
enum ReadRuleType {
    READ_FROM_BEGINNING = 0;
    READ_FROM_END = 1;
}

// the next will be 2
enum WriteRuleType {
    WRITE_FROM_BEGINNING = 0;
    WRITE_FROM_END = 1;
}

// the next will be 5
enum PartitionerStorageType {
    YEARLY = 0;
    MONTHLY = 1;
    DAILY = 2;
    HOURLY = 3;
    MINUTELY = 4;
}

// the next will be 5
enum Status {
    IDLE = 0;
    WAITING = 1;
    RUNNING = 2;
    SUCCEEDED = 3;
    FAILED = 4;
}

// the next will be 2
enum Signal {
    START = 0;
    STOP = 1;
}
